tox_extension_messages.c

```diff
@@ -259,11 +432,15 @@ static void tox_extension_messages_neg(struct ToxExtExtension *extension,
 				       struct ToxExtPacketList *response_packet)
 {
 	(void)extension;
-	(void)response_packet;
-	struct ToxExtensionMessages *ext_message_ids = userdata;
-	get_or_insert_incoming_message(ext_message_ids, friend_id);
-	ext_message_ids->negotiated_cb(friend_id, compatible,
-				       ext_message_ids->userdata);
+	struct ToxExtensionMessages *ext_messages = userdata;
+	get_or_insert_incoming_message(ext_messages, friend_id);
+	if (!compatible) {
+		ext_messages->negotiated_cb(friend_id, compatible,
+					    ext_messages->userdata);
+	} else {
+		tox_extension_messages_negotiate_size(ext_messages,
```

Comment:
I can't remember the design here, but I think it's worth a double check if clients are expecting negotiation to timeout. If toxext gives up at some point and sends a false compatible flag this could introduce a new case when the user goes offline between negotiation and receiving the size negotiation packet.

Reply:
It looks like toxext only calls the negotiation callback with false  when revoking or when it gets an explicit negotiation response saying the friend doesn't support the extension, so I don't think we're at risk of toxext timing out then us stalling the callback to clients forever.